plugins/hls-tactics-plugin/src/Wingman/Machinery.hs

```diff
@@ -342,7 +342,7 @@ lookupNameInContext name = do
 getDefiningType
     :: TacticsM CType
 getDefiningType = do
-  calling_fun_name <- fst . head <$> asks ctxDefiningFuncs
+  calling_fun_name <- asks (fst . headNote "Wingman.Machinery.getDefiningType" . ctxDefiningFuncs)
```

Comment:
Just below we do something with `failure`, can we also use that branch if we fail here?